test/rules/test_md050.py

```diff
@@ -0,0 +1,201 @@
+"""
+Module to provide tests related to the MD050 rule.
+"""
+import os
+from test.markdown_scanner import MarkdownScanner
+from test.utils import write_temporary_configuration
+
+import pytest
+
+
+@pytest.mark.rules
+def test_md050_refs_should_only_contain_numbers():
+    # Arrange
+    scanner = MarkdownScanner()
+    source_path = os.path.join("test", "resources", "rules", "md050", "only-letters.md")
+    supplied_configuration = {
+        "log": {"level": "INFO", "stack-trace": True},
+        "plugins": {
+            "md050": {
+                "enabled": True,
+                "regexes": {
+                    "1": {
+                        "regex": "[0-9]+",
+                        "errorMessage": "Should only contain numbers",
+                    }
+                },
+            }
+        },
+    }
+    configuration_file = None
+    try:
+        configuration_file = write_temporary_configuration(supplied_configuration)
+        supplied_arguments = [
+            "-c",
+            configuration_file,
+            "-e md050",
+            "scan",
+            source_path,
+        ]
+
+        expected_return_code = 1
+        expected_output = f"{source_path}:1:1: MD050: Reference should match regular expressions [Should only contain numbers] (refs-regex)"
+        expected_error = ""
+
+        # Act
+        execute_results = scanner.invoke_main(arguments=supplied_arguments)
+
+        # Assert
+        execute_results.assert_results(
+            expected_output, expected_error, expected_return_code
+        )
+    finally:
+        if configuration_file and os.path.exists(configuration_file):
+            os.remove(configuration_file)
+
+
+@pytest.mark.rules
+def test_md050_refs_should_be_internal():
+    # Arrange
+    scanner = MarkdownScanner()
+    source_path = os.path.join(
+        "test", "resources", "rules", "md050", "external-reference.md"
+    )
+    message = "Links should be internal."
+    supplied_configuration = {
+        "log": {"level": "INFO", "stack-trace": True},
+        "plugins": {
+            "md050": {
+                "enabled": True,
+                "regexes": {
+                    "1": {
+                        "regex": r"^(https?:)?(\/\/)?git\.t3\.daimlertruck\.com\/EWOE-PD\/TruckBOM-Doku\/.*$",
```

Comment:
We want to create a upstream PR for this rule. I would recommend to not put customer data into the test code.